Flat/src/ModulesLepPho.cc

```diff
@@ -9,139 +9,262 @@
 using namespace panda;
 using namespace std;
 
+void PandaAnalyzer::SimpleLeptons() {
+  looseLep1PdgId=-1, looseLep2PdgId=-1;
+  //electrons
+  for (auto& ele : event.electrons) {
+    float pt = ele.pt(); float eta = ele.eta(); float aeta = fabs(eta);
+    if (pt<10 || aeta>2.5) continue;
+    if (!ele.veto) continue;
+    if (!ElectronIP(ele.eta(),ele.dxy,ele.dz)) continue;
+    ele.setPtEtaPhiM(pt,eta,ele.phi(),511e-6);
+    unsigned iL=gt->nLooseElectron;
+    bool isFake   = ele.hltsafe;
+    bool isMedium = ele.medium;
+    bool isTight  = ele.tight;
+    if (isTight) gt->nTightElectron++;
+    int eleSelBit            = kLoose;
+    if (isFake  ) eleSelBit |= kFake;
+    if (isMedium) eleSelBit |= kMedium;
+    if (isTight ) eleSelBit |= kTight;
+    gt->electronPt[iL]           = pt;
+    gt->electronEta[iL]          = eta;
+    gt->electronPhi[iL]          = ele.phi();
+    gt->electronSfLoose[iL]      = GetCorr(cEleLoose, eta, pt);
+    gt->electronSfTight[iL]      = GetCorr(cEleTight, eta, pt);
+    gt->electronSfUnc[iL]        = GetError(cEleMedium, eta, pt);
+    gt->electronSfReco[iL]       = GetCorr(cEleReco, eta, pt);
+    gt->electronSelBit[iL]       = eleSelBit;
+    gt->electronPdgId[iL]        = ele.charge*-11;
+    gt->electronTripleCharge[iL] = ele.tripleCharge;
+    looseLeps.push_back(&ele);
+    matchLeps.push_back(&ele);
+    matchEles.push_back(&ele);
+    gt->nLooseElectron++;
+    if (gt->nLooseElectron>=2) break;
+  }
+  // muons
+  for (auto& mu : event.muons) {
+    float pt = mu.pt(); float eta = mu.eta(); float aeta = fabs(eta);
+    if (pt<10 || aeta>2.4) continue;
+    if (!mu.loose) continue;
+    mu.setPtEtaPhiM(pt,eta,mu.phi(),0.106);
```

Comment:
Uh why are we changing the 4-vector of a muon...?

Reply:
Ok, so this came from the implementation in ComplicatedLeptons which changes the muon pT.
Since SimpleLeptons does not correct the muon pT, I agree we can drop that line